packages/input-tel/test-suites/LionInputTel.suite.js

```diff
@@ -0,0 +1,157 @@
+import {
+  expect,
+  fixture as _fixture,
+  html,
+  defineCE,
+  unsafeStatic,
+  aTimeout,
+} from '@open-wc/testing';
+import { localize } from '@lion/localize';
+import { LionInputTel } from '../src/LionInputTel.js';
+import { IsPhoneNumber } from '../src/validators.js';
+import { LibPhoneNumber } from '../src/LibPhoneNumber.js';
+import { mockLibPhoneNumber, restoreLibPhoneNumber } from '../test-helpers/mockLibPhoneNumber.js';
+
+/**
+ * @typedef {import('@lion/core').TemplateResult} TemplateResult
+ */
+const fixture = /** @type {(arg: string | TemplateResult) => Promise<LionInputTel>} */ (_fixture);
+
+const getCountryCodeBasedOnLocale = () =>
+  // @ts-expect-error [allow-protected]
+  localize._getLangFromLocale(localize.locale).toUpperCase();
+
+/**
+ * @param {{ klass:LionInputTel }} config
+ */
+// @ts-ignore
+export function runInputTelSuite({ klass = LionInputTel } = {}) {
+  // @ts-ignore
+  const tagName = defineCE(/** @type {* & HTMLElement} */ (class extends klass {}));
+  const tag = unsafeStatic(tagName);
+
+  describe('LionInputTel', () => {
+    beforeEach(async () => {
+      // Wait till LibPhoneNumber has been loaded
+      await LibPhoneNumber.loadComplete;
+    });
+
+    describe('Country code', () => {
+      it('automatically localizes based on current locale', async () => {
+        const el = await fixture(html` <${tag}></${tag}> `);
+        const currentCode = getCountryCodeBasedOnLocale();
+        expect(el.countryCode).to.equal(currentCode);
+      });
+
+      it('can preconfigure the country code via attr', async () => {
+        const currentCode = getCountryCodeBasedOnLocale();
+        const newCode = currentCode === 'DE' ? 'NL' : 'DE';
+        const el = await fixture(html` <${tag} country-code="${newCode}"></${tag}> `);
+        expect(el.countryCode).to.equal(newCode);
+      });
+
+      it('can preconfigure the country code via prop', async () => {
+        const currentCode = getCountryCodeBasedOnLocale();
+        const newCode = currentCode === 'DE' ? 'NL' : 'DE';
+        const el = await fixture(html` <${tag} .countryCode="${newCode}"></${tag}> `);
+        expect(el.countryCode).to.equal(newCode);
+      });
+
+      it('reformats when country code is changed on the fly', async () => {
+        const el = await fixture(
+          html` <${tag} .countryCode="${'NL'}" .modelValue="${'612345678'}" ></${tag}> `,
+        );
```

Comment:
Can you also enter `+31 6 12345678`?

What happens if `countyCode` and the country phone code (for NL it's `+31`/`0031`) don't match up?

Reply:
You can actually. Ideally, we should maybe restrict to only accept e164 as modelValue/serializedValue. Also see: https://github.com/ing-bank/lion/issues/1628
Wrt view values/ manually entering, it's interpreted by awesome-phonenumber based on .formatStrategy.